src/parser/smt2/smt2.cpp

```diff
@@ -240,14 +240,23 @@ void Smt2::addTheory(Theory theory) {
     addOperator(kind::INSERT, "insert");
     addOperator(kind::CARD, "card");
     addOperator(kind::COMPLEMENT, "complement");
+    addOperator(kind::JOIN, "join");
+    addOperator(kind::PRODUCT, "product");
+    addOperator(kind::TRANSPOSE, "transpose");
+    addOperator(kind::TCLOSURE, "tclosure");
     break;
 
   case THEORY_DATATYPES:
+  {
+    const std::vector<Type> types;
+    const std::vector<Type>& types2 = types;
+    defineType("Tuple", getExprManager()->mkTupleType(types2));
```

Comment:
Why not pass `types` here instead of `types2`?

Reply:
I seemed to have a typing error when I tried to do that, but it seems to work now, so I must have missed something. I have changed it now to pass 'types' to mkTupleType() instead. Thanks.